mautic/mautic
# Decode equals sign while decoding tokens fix

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ✅
| New feature/enhancement? (use the a.x branch)      | ✅
| Deprecations?                          | ❌
| BC breaks? (use the c.x branch)        | ❌
| Automated tests included?              | ✅ 
| Related user documentation PR URL      |  
| Related developer documentation PR URL |  
| Issue(s) addressed                     | 



#### Description:

Mautic code is taking the URL encoded tokens into account when it comes to character like "{" and "}". But "=" characters are not taken into account by Mautic code base. This PR addresses the issue.

#### Steps to test this PR:


1. Make sure you have at least one contact with email and last name filled in.
2. Create a new template email with the required fields filled in.
3. Enter the email builder.
4. Add a new icon by dragging the `Icons` tile from the right pane.
5. Click the `Add new icon` button.
6. Click the `More options` toggle.
7. Change the `Link type` to `Send email`.
8. Fill in `Mail to` with `{contactfield=email}`
9. Fill in `Subject` with `{contactfield=lastname}`
10. The builder view should look like this:
  
11. Close the builder.
12. Save the email by clicking the `Save & close` button.
13. Fill in the `Show preview for contact` field with the contact from the 1. step.
14. Click the preview button.
15. Hover the icon.
16. You should see both tokens replaced.
 
17. Make sure the tokens are also replaced when a real email is sent.


#### Other areas of Mautic that may be affected by the change:
1. All places where tokens are replaced for emails.

[//]: # ( As applicable: )
#### List of areas covered by the unit and/or functional tests:
1. Email tokens replacement.
